crates/transaction-pool/src/pool/mod.rs

```diff
@@ -232,13 +232,19 @@ where
     /// Returns hashes of _all_ transactions in the pool.
     pub(crate) fn pooled_transactions_hashes(&self) -> Vec<TxHash> {
         let pool = self.pool.read();
-        pool.all().hashes_iter().collect()
+        pool.all()
+            .hashes_iter()
+            .filter(|hash| match pool.get(hash) {
+                Some(tx) => tx.propagate,
+                None => false,
+            })
+            .collect()
```

Comment:
this is not really performant because this now adds another lookup per item

instead of hashes iter, we should iterate over the tx objects

Reply:
How should I resolve the resulting warning from hashes_iter being unused?

Reply:
marked it as allow unused, perhaps still useful